crates/meilisearch-types/src/settings.rs

```diff
@@ -134,79 +144,142 @@ impl MergeWithError<milli::CriterionError> for DeserrJsonError<InvalidSettingsRa
     }
 }
 
+#[derive(Default, Serialize, Deserialize, PartialEq, Eq, Clone, ToSchema)]
+#[repr(transparent)]
+#[serde(transparent)]
+pub struct SettingEmbeddingSettings {
```

Comment:
can you elaborate as a response to this comment why the intermediate struct is needed?